builder/builder-next/builder.go

```diff
@@ -644,3 +660,31 @@ func toBuildkitPruneInfo(opts types.BuildCachePruneOptions) (client.PruneInfo, e
 		Filter:       []string{strings.Join(bkFilter, ",")},
 	}, nil
 }
+
+// sanitizeRepoAndTags parses the raw "t" parameter received from the client
+// to a slice of repoAndTag. It removes duplicates and validates each repoName
+// and tag to not contain a digest.
+func sanitizeRepoAndTags(names []string) (repoAndTags []string, err error) {
+	uniqNames := map[string]struct{}{}
+	for _, repo := range names {
+		if repo == "" {
+			continue
+		}
+
+		ref, err := reference.ParseNormalizedNamed(repo)
+		if err != nil {
+			return nil, err
+		}
+
+		if _, ok := ref.(reference.Digested); ok {
+			return nil, errors.New("build tag cannot contain a digest")
+		}
+
+		nameWithTag := reference.TagNameOnly(ref).String()
+		if _, exists := uniqNames[nameWithTag]; !exists {
+			uniqNames[nameWithTag] = struct{}{}
+			repoAndTags = append(repoAndTags, nameWithTag)
+		}
+	}
+	return repoAndTags, nil
+}
```

Comment:
Given we now have this function verbatim in two packages, is there a common place they could maybe both live as an exported function instead? :sweat_smile:

Reply:
done 👍